### PR TITLE
feat: simplify to 3 modes (builder/analyst/plan) + SQL write access control

### PR DESCRIPTION
**Pull request by kulvirgit**
## Summary
- Remove 5 agents (executive, validator, migrator, researcher, trainer) — simplify to builder, analyst, plan
- Add SQL write access control: regex-based classifier detects write queries, enforces via new `sql_execute_write` permission
- Analyst mode: truly read-only — unknown bash denied, safe commands auto-allowed, dbt write commands denied, SQL writes denied
- Builder mode: SQL writes prompt for approval, DROP DATABASE/SCHEMA/TRUNCATE hard-blocked
- Defense-in-depth: hardcoded safety blocks in sql-execute.ts + permission system

## Changes
- `agent.ts`: Remove 5 agent definitions, update analyst/builder permissions
- `sql-classify.ts`: New SQL write classifier (INSERT, UPDATE, DELETE, DROP, CREATE, ALTER, TRUNCATE, MERGE, GRANT, REVOKE, COPY INTO, CALL, EXEC, BEGIN, DECLARE, EXECUTE IMMEDIATE)
- `sql-execute.ts`: Add classification + permission check before execution
- Delete 5 prompt files (executive, validator, migrator, researcher, trainer)
- Clean up references in tests, memory budgets, verify-restructure

## Test plan
- [x] 47 SQL classifier tests (read/write/edge cases including CTEs, nested parens, multi-statement, BEGIN blocks)
- [x] 36 agent tests pass
- [x] TypeScript compiles
- [ ] Manual: TUI shows only builder/analyst/plan
- [ ] Manual: analyst can't run INSERT INTO, builder prompts for it

## Review
Reviewed by 7 AI models (Claude, Gemini, Kimi, Grok, MiniMax, GLM-5, Qwen). Key fixes from review:
- CTE regex bypass (nested parens) → fixed with paren-depth counting
- HARD_DENY multi-statement bypass → fixed by splitting on semicolons
- BEGIN/DECLARE block bypass → added to write keywords

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added data-visualization skill (/data-viz) for interactive dashboards and charts from query results

* **Behavior Changes**
  * Builder now requires explicit approval for SQL writes; Analyst is enforced read-only for SQL writes
  * Non-overridable safety denials block destructive SQL (DROP DATABASE/SCHEMA, TRUNCATE)

* **Documentation**
  * Clarified agent capability docs and wording around plan/analyst restrictions

* **Tests**
  * Added comprehensive SQL-classification and permission tests

* **Chores**
  * Removed several legacy agent prompt modes from defaults

### CODE REVIEW

**Review by suryaiyer95 — COMMENTED** [review]
Some comments and sentry comments

File: packages/opencode/src/altimate/tools/sql-classify.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/**
+ * Single-pass: classify and check for hard-denied statement types.
+ * Returns both the overall query type and whether a hard-deny pattern was found.
+ */
+export function classifyAndCheck(sql: string): { queryType: "read" | "write"; blocked: boolean } {
+  const result = core.getStatementTypes(sql)
+  if (!result?.statements?.length) return { queryType: "read", blocked: false }
+
+  const blocked = result.statements.some((s: { statement_type: string }) =>
+    HARD_DENY_TYPES.has(s.statement_type.toUpperCase()),
```
[the remark this answers is not in the document]
Comment (kulvirgit) [reply packages/opencode/src/altimate/tools/sql-classify.ts]: **Re: Other/unknown categories**

Good call. Right now `getStatementTypes()` returns these categories: `dml`, `ddl`, `dcl`, `tcl`, and `other`. 

- `other` covers read-only metadata ops like `SHOW`, `EXPLAIN`, `SET`, `USE` — safe to auto-allow.
- If the parser returns an **unrecognized** category (not in any of our sets), the current code defaults to `"read"` and lets it through silently.

I'll change the logic so that unknown/unrecognized categories (anything not in `WRITE_CATEGORIES` and not `"other"`) trigger the `sql_execute_write` permission prompt — same as a write query. That way if altimate-core adds a new category we don't know about, we fail safe by asking the user rather than silently allowing.

---

**Re: regex approach**

No — regex is fully removed. The classifier is 100% AST-based via `altimate-core.getStatementTypes()` now. 

The old regex approach had the bypass issues sentry-bot flagged (CTE parsing, string literals, comment stripping). The AST parser handles all of those correctly since it actually parses the SQL rather than pattern-matching on the surface text.

The only "pattern matching" left is the `HARD_DENY_TYPES` set which checks the `statement_type` strings returned by the AST (e.g. `"DROP DATABASE"`, `"TRUNCATE TABLE"`) — but that's matching on parsed AST output, not raw SQL text.
